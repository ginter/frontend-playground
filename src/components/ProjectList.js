import React from 'react'

const ProjectList = ({ projects, deleteProject }) => {
  if (projects.length === 0) return <p>No projects found</p>;

  return(
    <ul>
      {projects.map(project => (
        <li key={project.uid} className='project-row'>
          {project.name}
          <button onClick={() => deleteProject(project.uid)}>DELETE</button>
        </li>
        )
      )}
    </ul>
  );
}

export default ProjectList
