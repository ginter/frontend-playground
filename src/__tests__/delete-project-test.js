import React from 'react'
import renderer from 'react-test-renderer'

import ProjectList from '../components/project-list'

jest.unmock('../components/project-list');

describe('ProjectList', () => {
  it('renders correctly with no projects', () => {
    const component = renderer.create(<ProjectList projects={[]} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
