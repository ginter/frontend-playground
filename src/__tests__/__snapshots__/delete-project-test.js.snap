exports[`ProjectList renders correctly with no projects 1`] = `
<p>
  No projects found
</p>
`;
